redis/commands/mset.cc

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "redis/commands/mset.hh"
+#include "seastar/core/shared_ptr.hh"
+#include "redis/request.hh"
+#include "redis/reply.hh"
+#include "service_permit.hh"
+#include "types.hh"
+#include "service/storage_proxy.hh"
+#include "service/client_state.hh"
+#include "redis/options.hh"
+#include "redis/mutation_utils.hh"
+
+namespace redis {
+
+namespace commands {
+
+shared_ptr<abstract_command> mset::prepare(service::storage_proxy& proxy, request&& req)
```

Comment:
We should document somewhere, at least as a comment here but probably also in redis.md (which should have a compatibility section like we have in docs/alternator/alternator.md), that this "mset" implementation offers no atomicity guarantee. https://redis.io/commands/mset explains that "all given keys are set at once. It is not possible for clients to see that some of the keys were updated while others are unchanged." - but this is not promised by this implementation.

Reply:
We are discussing the atomic issue by email.

Do you prefer to find a solution first?  or we can implement commands without care about the atomicity guarantee (document it somewhere, and fix it in future)?

Reply:
I don't know how we can provide this guarantee at all, since in Scylla, the different keys can be in different nodes, and we would need full transaction support to implement this. So let's just start with documenting the problem (as you did), and fix it (much) later.